internal/credit/postgres_connector/reset.go

```diff
@@ -20,6 +20,11 @@ type resetWithRollovedGrants struct {
 func (c *PostgresConnector) Reset(ctx context.Context, reset credit.Reset) (credit.Reset, []credit.Grant, error) {
 	ledgerID := credit.NewNamespacedLedgerID(reset.Namespace, reset.LedgerID)
 
+	// Truncate EffectiveAt to the window size
+	if effAt := reset.EffectiveAt.Truncate(c.config.WindowSize); effAt != reset.EffectiveAt {
```

Comment:
Given we have already calculated the truncated value we can just set it without the if.

```suggestion
	reset.EffectiveAt = reset.EffectiveAt.Truncate(c.config.WindowSize)
```



Also can you add a bit more verbose comment here why we are doing this? (e.g. clickhouse limitations, so that if somebody touches the code they have a better understanding).